ui/src/StoreProviderOrNull.jsx

```diff
@@ -0,0 +1,27 @@
+import { Provider } from 'react-redux';
+import React, { useState, useEffect } from 'react';
+import createStore from './store';
+
+import { reHydrateRootStateFromStorage } from './middlewares/statePersister';
+
+export const StoreProviderOrNull = ({ children }) => {
+	const [isBootstrapped, setBootstrapped] = useState(false);
+	const [data, setData] = useState(false);
+	
+	const reducersToPersist = ['ui', 'user'];
```

Comment:
we have this in other places as well, let's add it somewhere to use it in both places